web/toast_manager.js

```diff
@@ -0,0 +1,52 @@
+class ToastManager {
+  #toastElement;
+
+  #duration;
+
+  #timeoutID = null;
+
+  #controller = null;
+
+  constructor(elem, duration) {
+    this.#toastElement = elem;
+    this.#duration = duration;
+  }
+
+  show(action, type) {
+    this.#toastElement.setAttribute("data-l10n-args", JSON.stringify({ type }));
+    this.#toastElement.removeAttribute("hidden");
+    this.#timeoutID = setTimeout(() => {
+      this.#hide();
+    }, this.#duration);
+    this.#controller = new AbortController();
+    console.log(this.#toastElement, this.#toastElement.getElementById);
+    this.#toastElement
+      .querySelector("#annotationRemovedUndoButton")
```

Comment:
Here, and elsewhere, please note that you're not "allowed" to lookup elements like this and compare with how e.g. the toolbars are handled.